src/ctl/src/cmd_impl/await_tree/analyze.rs

```diff
@@ -0,0 +1,185 @@
+    /// the bottleneck actor is still yielding output to downstream actors. A typical
+    /// case is JOIN amplification. So the corresponding actors are actively processing
+    /// the data but the EPOCH span is blocked.
+    pub fn is_bottleneck(&self) -> bool {
+        fn visit(node: &SpanNodeView) -> bool {
+            let elapsed_secs = node.elapsed_ns as f64 / 1_000_000_000.0;
+            let slow_span = !node.span.is_long_running && elapsed_secs >= 10.0;
+            let is_epoch = node.span.name.starts_with("Epoch");
+
+            if !is_epoch && !node.children.is_empty() {
+                // IB Tree's `Epoch` span may have a long elapsed time, though it's not
+                // a bottleneck. We exclude the `Epoch` span from the bottleneck detection
+                let mut elapsed_sum = 0.;
+                let mut elapsed_count = 0;
+                for child in &node.children {
+                    elapsed_count += 1;
+                    elapsed_sum += child.elapsed_ns as f64 / 1_000_000_000.0;
+                }
+                let elapsed_avg = elapsed_sum / elapsed_count as f64;
+                if slow_span && (elapsed_avg * 5.0 < elapsed_secs) {
+                    return true;
+                }
+            }
+
+            // visit children recursively
+            for child in &node.children {
+                if visit(child) {
+                    return true;
+                }
+            }
+            false
+        }
+        visit(&self.tree)
+    }
+}
+
+pub async fn bottleneck_detect(context: &CtlContext, path: Option<String>) -> anyhow::Result<()> {
+    if let Some(path) = path {
+        bottleneck_detect_from_file(path)
+    } else {
+        bottleneck_detect_real_time(context).await
+    }
+}
+
+async fn bottleneck_detect_real_time(context: &CtlContext) -> anyhow::Result<()> {
+    let meta_client = context.meta_client().await?;
+
+    let compute_nodes = meta_client
+        .list_worker_nodes(Some(WorkerType::ComputeNode))
+        .await?;
+    let clients = ComputeClientPool::adhoc();
+
+    let req = StackTraceRequest::default();
+
+    let mut bottleneck_actors_found = false;
+    for cn in compute_nodes {
+        let client = clients.get(&cn).await?;
+        let response = client.stack_trace(req.clone()).await?;
+        for (actor_id, trace) in response.actor_traces {
+            let tree: TreeView = serde_json::from_str(&trace)
+                .map_err(|e| anyhow::anyhow!("Failed to parse actor trace JSON: {}", e))?;
+            if tree.is_bottleneck() {
+                bottleneck_actors_found = true;
+                println!(">> Actor {}", actor_id);
+                println!("{}", tree);
+            }
+        }
+    }
+
+    if !bottleneck_actors_found {
+        println!("No bottleneck actors detected.");
+    }
+
+    Ok(())
+}
+
+fn bottleneck_detect_from_file(path: String) -> anyhow::Result<()> {
+    let actor_traces = extract_actor_traces(&path)
```

Comment:
hmmm, is `extract_actor_traces` for TEXT only, and should not be applied to JSON?

Reply:
`extract_actor_traces` is for both JSON and TEXT. For either JSON or TEXT, the actor traces are encoded as 
```
--- Actor Traces ---
>> Actor {actor_id}
{trace}

...
```
So we can use one unified funtion to extract  `extract_actor_traces` from the diagnose file.